packages/core/src/storages/request_provider.ts

```diff
@@ -445,6 +445,13 @@ export abstract class RequestProvider implements IStorage {
             this._maybeAddRequestToQueueHead(request.id, forefront);
         }, STORAGE_CONSISTENCY_DELAY_MILLIS);
 
+        // Finally, try to delete the request lock (even if the request is not using locks)
```

Comment:
> even if the request is not using locks

why? is there actual use case where this is needed or is it just a failsafe?

Reply:
Welll...
- we plan on making v2 default once its stable enough
- overriding the method just to: execute super, store return, run something on the request then return the stored result; felt wrong (but can do it if its more desired)

Reply:
I agree with @B4nan that deleting the lock only in case we use the lock looks cleaner.

Reply:
Fair nuff, will adjust accordingly